core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -3006,12 +3005,22 @@ private RecordCursorFactory generateTableQuery(
                     }
                 }
 
+                boolean isOrderByTimestampDesc = isOrderDescendingByDesignatedTimestampOnly(model);
+                RowCursorFactory rowFactory;
+
+                if (isOrderByTimestampDesc && !intrinsicModel.hasIntervalFilters()) {
```

Comment:
It's a valid workaround, though it would be good if the optimisation works with interval filters. Most of queries on decent data sizes will have interval filters.

Reply:
I agree 100% and even started trying to implement bwd scan for other cursor types in the method but scope turned out too big . Each of the remaining cursor types will need a separate discussion and PR.